service/storage_proxy.cc

```diff
@@ -5415,9 +5415,12 @@ storage_proxy::query_partition_key_range(lw_shared_ptr<query::read_command> cmd,
     schema_ptr schema = local_schema_registry().get(cmd->schema_version);
     replica::keyspace& ks = _db.local().find_keyspace(schema->ks_name());
 
-    // when dealing with LocalStrategy keyspaces, we can skip the range splitting and merging (which can be
-    // expensive in clusters with vnodes)
-    query_ranges_to_vnodes_generator ranges_to_vnodes(get_token_metadata_ptr(), schema, std::move(partition_ranges), ks.get_replication_strategy().get_type() == locator::replication_strategy_type::local);
+    // when dealing with LocalStrategy and EverywhereStrategy keyspaces, we can skip the range splitting and merging
+    // (which can be expensive in clusters with vnodes)
+    auto merge_tokens = ks.get_replication_strategy().get_type() == locator::replication_strategy_type::local ||
+                        ks.get_replication_strategy().get_type() == locator::replication_strategy_type::everywhere_topology;
```

Comment:
@bhalevy IIRC you added a method that determines this

Reply:
right.  that's `!ks.get_replication_strategy().natural_endpoints_depend_on_token()`